ui/pages/providers/[id].js

```diff
@@ -1,14 +1,130 @@
 import { useRouter } from 'next/router'
+import copy from 'copy-to-clipboard'
 import Head from 'next/head'
 import Link from 'next/link'
 import useSWR, { useSWRConfig } from 'swr'
+import { DuplicateIcon, CheckIcon } from '@heroicons/react/outline'
 import dayjs from 'dayjs'
 
 import Dashboard from '../../components/layouts/dashboard'
 import RemoveButton from '../../components/remove-button'
 import Notification from '../../components/notification'
 
+function SCIMKeyDialog(props) {
+  const [scimAccessKey, setSCIMAccessKey] = useState('')
+  const [error, setError] = useState('')
+  const [keyCopied, setKeyCopied] = useState(false)
+
+  async function onSubmit(e) {
+    e.preventDefault()
+
+    try {
+      let keyName = props.provider.name + '-scim'
+
+      // delete any existing access key for this provider
+      await fetch(`/api/access-keys?name=${keyName}`, {
+        method: 'DELETE',
+      })
+
+      // generate the new key
+      const res = await fetch('/api/access-keys', {
+        method: 'POST',
+        body: JSON.stringify({
+          userID: props.provider.id,
+          name: keyName,
+          ttl: '87600h',
+          extensionDeadline: '720h',
+        }),
+      })
+
+      const data = await jsonBody(res)
+
+      setSCIMAccessKey(data.accessKey)
+    } catch (e) {
+      setError(e.message)
+    }
+  }
+
+  return (
+    <div className='w-full 2xl:m-auto'>
+      <h1 className='py-1 font-display text-lg font-medium'>SCIM Access Key</h1>
+      <div className='space-y-4'>
+        {scimAccessKey === '' && error === '' ? (
+          <section>
+            <form onSubmit={onSubmit} className='flex flex-col space-y-4'>
+              <div className='mb-4 flex flex-col'>
+                <div className='relative mt-4'>
+                  <h2 className='mt-5 text-sm'>
+                    Generating a new SCIM access key will revoke any existing
+                    SCIM access key for this identity provider.
+                  </h2>
+                  <h2 className='mt-5 text-sm'>Do you wish to continue?</h2>
+                </div>
+              </div>
+              <div className='flex flex-row items-center justify-end space-x-3'>
+                <button
+                  type='button'
+                  onClick={() => props.setOpen(false)}
+                  className='inline-flex items-center rounded-md border border-gray-300 bg-white px-4 py-2 text-xs font-medium text-gray-700 shadow-sm hover:bg-gray-100'
+                >
+                  Cancel
+                </button>
+                <button
+                  type='submit'
+                  className='inline-flex items-center rounded-md border border-transparent bg-black px-4 py-2 text-xs font-medium text-white shadow-sm hover:bg-gray-800'
+                >
+                  Continue
+                </button>
+              </div>
+            </form>
+          </section>
+        ) : (
+          <>
+            <section>
```

Comment:
we should be able to refactor this to make it reusable since we have the same function in `../add.js` too